app/lib/search/text_utils.dart

```diff
@@ -110,8 +99,43 @@ Map<String, double> tokenize(String originalText) {
   return tokens;
 }
 
+// Scan for CamelCase phrases and extract Camel and Case separately.
+Iterable<String> extractCamelCase(String word) sync* {
+  final changeIndex = <int>[0];
+  bool prevLower = _isLower(word[0]);
+  for (int i = 1; i < word.length; i++) {
+    final bool lower = _isLower(word[i]);
+    if (!lower && prevLower) {
+      changeIndex.add(i);
+    }
+    prevLower = lower;
+  }
+  changeIndex.add(word.length);
+  for (int i = 1; i < changeIndex.length; i++) {
+    final token = word.substring(changeIndex[i - 1], changeIndex[i]);
+    yield token;
+  }
+}
+
 bool _isLower(String c) => c.toLowerCase() == c;
 
+/// Returns whether an API symbol can be considered as self-documenting.
+///
+/// When an API symbol uses common english words for most of its parts, we
+/// assume that it is descriptive enough for most use-case.
+bool isSelfDocumenting(String symbol) {
+  final words = extractCamelCase(symbol).map((s) => s.toLowerCase()).toSet();
+  final englishLength = words
+      .where(
+        (word) =>
+            english_words.all.contains(word) ||
```

Comment:
I think `english_words.all` is a list. Probably this would be faster if you convert it to a Set<String> first.